src/main.rs

```diff
@@ -351,6 +382,25 @@ impl HunkBuffer {
                             &mut shared_removed,
                         )
                     };
+
+                    if config.line_numbers {
+                        let mut margin_buf = &mut margin[..];
+                        if is_plus {
```

Comment:
minor: I think you can merge the two branches with something like that (unchecked):
```
//  margin is filled with `b' '`  initially
let ofs = if current_line_minus != 0 {
    write(&mut margin, "{}", current_line_minus)?;
    half_margin + 1
} else {
    0
};
if current_line_plus != 0 {
    write(&mut margin[ofs..], "{}", current_line_minus)?;
}
```

Reply:
This is nicest I could make the final code look like with this approach, I think it ends up being harder to follow in my opinion.

```rust
for v in &mut margin[..] { // Is there a more concise way to clear/fill a slice?
    *v = b' ';
}
// Renaming the current_line_ prefix to lino_ so rustfmt doesn't wrap the lines
let ofs = if lino_minus != 0 {
    if !is_plus {
        write!(&mut margin[..], "{:w$}", lino_minus, w = half_margin)?;
        lino_minus += 1;
    }
    half_margin + 1
} else {
    0
};

if is_plus && lino_plus != 0 {
    write!(&mut margin[ofs..], "{:w$}", lino_plus, w = half_margin)?;
    lino_plus += 1;
}
```